### PR TITLE
Adding a deprecation info API check for saml nameid_format setting

### PR DESCRIPTION
**Pull request by masseyke**
In 8.0 the behavior of an unset saml nameid_format setting. This commit adds a check to the deprecation
info API that warns users of this change, if they are using saml without nameid_format being set.
Relates #42404 #44090

### CODE REVIEW

File: x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecksTests.java [hunk 1]
```diff
@@ -1404,4 +1402,69 @@ public void testCheckMaxLocalStorageNodesSetting() {
...
+
+        settings = Settings.builder()
+            .put("xpack.security.authc.realms.saml.saml1.attributes.principal", randomIntBetween(30, 100))
+            .put("xpack.security.authc.realms.saml.saml2.attributes.principal", randomIntBetween(30, 100))
+            .put("xpack.security.authc.realms.saml.saml2.nameid_format", randomIntBetween(1, 25))
+            .build();
+        expectedIssue = new DeprecationIssue(DeprecationIssue.Level.WARNING,
+            "if nameid_format is not explicitly set, the previous default of 'urn:oasis:names:tc:SAML:2.0:nameid-format:transient' is no " +
+                "longer used",
+            "https://www.elastic.co/guide/en/elasticsearch/reference/master/saml-guide.html",
```
Comment (jkakavas) [inline x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecksTests.java:1444]: I think we either need some more text here or a section in the breaking release notes and link from here ( I think the latter gives us more flexibility ). I can open a PR for the docs and reference the logs here once we have a link

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java [hunk 2]
```diff
@@ -970,4 +971,36 @@ static DeprecationIssue checkMaxLocalStorageNodesSetting(final Settings settings
...
+                                concreteNameIdFormatSettingKey, realm);
+                        }
+                    }
+                    return null;
+                })
+                .filter(detail -> detail != null).collect(Collectors.toList());
+        if (detailsList.isEmpty()) {
+            return null;
+        } else {
+            String message = "if nameid_format is not explicitly set, the previous default of " +
```
Comment (jkakavas) [inline x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java:999]: I think we either need some more text here or a section in the breaking release notes and link from here ( I think the latter gives us more flexibility ). I can open a PR for the docs and reference the logs here once we have a link

**Comment by masseyke** [comment]
@elasticmachine update branch

**Comment by masseyke** [comment]
@elasticmachine update branch

**Review by jbaiera — APPROVED** [review]
LGTM, pending results of the discussion about the docs

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java (hunk 2 above)
Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java:1027]: Hi @jkakavas -- just confirming that this is still on your to-do list, and you're not waiting on anything from me.

Comment (jkakavas) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java:1027]: Thank you for the ping, you are right, this slipped my mind. I'll work on something today or tomorrow and get @lockewritesdocs to review.

Comment (jkakavas) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java:1027]: Here is it: https://github.com/elastic/elasticsearch/pull/77785

Comment (masseyke) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java:1027]: Thanks. I'll go ahead and merge this, and update its URL along with several others in an upcoming PR.

**Pull request merged**